lib/Instrumentation/AMDGPUHello.cpp

```diff
@@ -0,0 +1,70 @@
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/Pass.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Support/raw_ostream.h"
+#include <iostream>
+#include <vector>
+using namespace llvm;
+using namespace std;
+
+namespace {
+
+struct AMDGPUHello : public PassInfoMixin<AMDGPUHello> {
+  PreservedAnalyses run(Module &M, ModuleAnalysisManager &) {
+    bool ModifiedCodeGen = runOnModule(M);
+
+    return (ModifiedCodeGen ? llvm::PreservedAnalyses::none()
+                    : llvm::PreservedAnalyses::all());
+  }
+  bool runOnModule(llvm::Module &M);
+  // isRequired being set to true keeps this pass from being skipped
+  // if it has the optnone LLVM attribute
+  static bool isRequired() { return true; }
+};
+
+} // end anonymous namespace
+
+bool AMDGPUHello::runOnModule(Module &M) {
+  bool ModifiedCodeGen = false;
+  auto &CTX = M.getContext();
+
+  for (auto &F : M) {
+    if (F.isIntrinsic())
+      continue;
+    if (F.getCallingConv() == CallingConv::AMDGPU_KERNEL) {
```

Comment:
Any reason this is AMD specific? Can we make it work in general? (And drop the `AMD` prefix throughout.)

Reply:
Drop AMD dependency and just made it "GPU kernel"